pkg/api/server/db/model.go

```diff
@@ -28,3 +32,32 @@ type Record struct {
 	Name       string `gorm:"index:records_by_name,priority:3"`
 	Data       []byte
 }
+
+// Annotations is a custom-defined type of a gorm model field.
+type Annotations map[string]string
+
+// Scan implements sql.Scanner interface, to let gorm know how to resolve the corresponding data from database.
+func (ann *Annotations) Scan(value interface{}) error {
+	if ann == nil {
+		return errors.New("nil pointer")
```

Comment:
If annotations is unset in the proto, will this cause an error? If so, I think we should allow that.

Let's make sure to add a test to verify the behavior.

Reply:
Done